src/sim_state_manager.rs

```diff
@@ -1,6 +1,8 @@
 use bevy::prelude::*;
 use bevy::math::Vec2;
-use anyhow::Result;
+use crate::error::Error;
```

Comment:
Why did we make this change?

Reply:
Using thiserror allows us to define errors for more specific error handling.